src/routing.ts

```diff
@@ -341,22 +366,17 @@ export function createRouterContext(
     }
   };
 
+  // Create a native transition, whenever the source changes from outside
   createRenderEffect(() => {
-    const { value, state } = source();
-    // Untrack this whole block so `start` doesn't cause Solid's Listener to be preserved
-    untrack(() => {
-      start(() => {
-        intent = "native";
-        if (value !== reference()) setReference(value);
-        setState(state);
-        resetErrorBoundaries();
-        submissions[1]([]);
-      }).then(() => {
-        intent = undefined;
-      });
-    });
+    const newSource = source();
+    if (newSource.value !== untrack(reference) || newSource.state !== untrack(state)) {
+      transition("native", newSource);
+    }
   });
 
+  // Is this needed?
+  //createRenderEffect(() => transition("initial", untrack(source)));
```

Comment:
nah this isn't needed, `initial` is only useful for loaders and is handled elsewhere